rfcs/text/0045-additional-vulnerability-fields.md

```diff
@@ -0,0 +1,138 @@
+- Date: **TBD** <!-- The ECS team sets this date at merge time. This is the date of the latest stage advancement. -->
+
+<!--
+As you work on your RFC, use the "Stage N" comments to guide you in what you should focus on, for the stage you're targeting.
+Feel free to remove these comments as you go along.
+-->
+
+<!--
+Stage 0: Provide a high level summary of the premise of these changes. Briefly describe the nature, purpose, and impact of the changes. ~2-5 sentences.
+-->
+This RFC proposes to expand the vulnerability fieldset to include more fields, the proposal takes into consideration various customer feedbacks provided to Security integration team, inputs from Infosec team managing vulnerabilities across Elastic and other companies. This will benefit our customers and internal product teams to provide more effective vulnerability management experience to end user. to come up with the list of fields, extensive research was done across various Vulnerability management products and schemas like OSV. It is a continuation of one of the previous RFC on similar topic- https://github.com/elastic/ecs/issues/1685
+
+<!--
+Stage 1: If the changes include field additions or modifications, please create a folder titled as the RFC number under rfcs/text/. This will be where proposed schema changes as standalone YAML files or extended example mappings and larger source documents will go as the RFC is iterated upon.
+-->
+
+<!--
+Stage X: Provide a brief explanation of why the proposal is being marked as abandoned. This is useful context for anyone revisiting this proposal or considering similar changes later on.
+-->
+
+## Fields
+The `vulnerabilities` fields being proposed are as follows:
+
+| Field | Type | Description / Use Case |
+| ----- | ---- | ---------------------- |
+| `vulnerability.vendor.id` | keyword | A vulnerability doesn't have necessary a CVE associated with it. I don't know if it makes sense to seperate vulnerability ID (like CVEs) to the vendor/detection IDs. |
+| `vulnerability.title`	| keyword | Title/Name/Short Description for vulnerability, to be used in flyout and dashboards. |
+| `vulnerability.mitigation` | text | Explains user how to fix or mitigate the problem, could be usefd to store resolution from the scanner vendor or document mitigation in place |
+| `vulnerability.published`	| date | The “published” field indicates the date when information about a specific vulnerability was publicly disclosed or made available.It represents the moment when details about the vulnerability were shared with the security community, vendors, and the public.This field helps security professionals track the timeline of vulnerability awareness.Example value: If a critical vulnerability (e.g., CVE-2024-1234) affecting a widely used software is disclosed on February 1, 2024, the published date for that CVE would be February 1, 2024. |
```

Comment:
could you add format of the date?

Reply:
Added format in example- Example value: If a critical vulnerability (e.g., CVE-2024-1234) affecting a widely used software is disclosed on February 1, 2024, the published date for that CVE would be 2024-02-01